entrypoint.py

```diff
@@ -83,6 +85,51 @@ async def pub_handler(request):
     return web.Response(text="success")
 
 
+# ws = new WebSocket("ws://${api_addr}/api/pub_ws")
+# ws.onopen = () => {
+#     ws.send(JSON.stringify({
+#         container: "...",
+#         topic: "...",
+#     }))
+# }
+# Then:
+# ws.send(JSON.stringify({
+#         container: "...",
+#         topic: "...",
+#         packet: {
+#             headers: [
+#                 ["key", "val"],
+#                 ...
+#             ],
+#             payload: window.btoa("..."),
+#         },
+#     }))
+async def pub_handler(request):
+    ws = web.WebSocketResponse()
+    await ws.prepare(request)
+
+    async for msg in ws:
+        if msg.type != WSMsgType.TEXT:
+            break
+
+        cmd = json.loads(msg.data)
+        global publishers
+
+        if cmd["topic"] in publishers:
```

Comment:
publishers should be keyed by container+topic

Reply:
Done.